### PR TITLE
added logic for user and partner deletion

### PR DESCRIPTION
**Pull request by reduno85**

### CODE REVIEW

File: user-partner-management-query/src/main/java/com/syncier/marketplace/userpartnermanagement/query/UserEntity.java [hunk 1]
```diff
@@ -51,4 +52,13 @@
     @OneToMany
     private Set<RoleEntity> roles;
 
+    @PreRemove
+    private void preRemove() {
+        if (partnerEntities != null) {
+            for (PartnerEntity partner : partnerEntities) {
+                partner.getUsers().remove(this);
+            }
```
Comment (msgbitcoder) [inline user-partner-management-query/src/main/java/com/syncier/marketplace/userpartnermanagement/query/UserEntity.java]: Don't delete the paren child relation that way. I prefer the delete cascade annotation on the relation. When you want to delete explicitly elements than put this implementation into the event handler implementation.

**Comment by reduno85** [comment]
@msgbitcoder  unit test are coming in a separate  branch  for the whole project

File: user-partner-management-query/src/main/java/com/syncier/marketplace/userpartnermanagement/query/UserEntity.java (hunk 1 above)
Comment (vsoltys) [reply user-partner-management-query/src/main/java/com/syncier/marketplace/userpartnermanagement/query/UserEntity.java]: Yes, please use JPA cascading instead

Comment (reduno85) [reply user-partner-management-query/src/main/java/com/syncier/marketplace/userpartnermanagement/query/UserEntity.java]: moved it to projector

**Pull request merged**